getoptions.sh

```diff
@@ -170,6 +170,7 @@ getoptions() {
   _2 'esac'
   _2 'shift'
   _1 'done'
+  _1 'unset OPTARG'
```

Comment:
The `OPTARG` variable should be available in the error handler to be able to generate friendly message. Therefore, it should be unset only when no errors. That is, include it on the next line.